### PR TITLE
8344229: Revisit SecurityManager usage in jdk.httpserver after JEP 486 integration

### PR DESCRIPTION
**Pull request by vy**
Removes `SecurityManager` and friends from `jdk.httpserver`. Successful `tier1..2` results are attached to the ticket.

**Methodology:** Case-insensitively and recursively searched for the occurrences of `("removal"|secu|privi)` in `httpserver`-named folders.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344229](https://bugs.openjdk.org/browse/JDK-8344229): Revisit SecurityManager usage in jdk.httpserver after JEP 486 integration (**Bug** - P4)

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22510/head:pull/22510` \
`$ git checkout pull/22510`

Update a local copy of the PR: \
`$ git checkout pull/22510` \
`$ git pull https://git.openjdk.org/jdk.git pull/22510/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22510`

View PR using the GUI difftool: \
`$ git pr show -t 22510`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22510.diff">https://git.openjdk.org/jdk/pull/22510.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22510#issuecomment-2514201848)
</details>

### CODE REVIEW

File: src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java [hunk 1]
```diff
@@ -55,91 +52,71 @@ class ServerConfig {
     private static long idleTimerScheduleMillis;
     private static long idleIntervalMillis;
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java:53]: while you're at it you could make these two fields final as well and use temporary variables in the static block to make sure they're assigned only once.

**Review by dfuch — COMMENTED** [review]
Good cleanup!

File: src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java [hunk 2]
```diff
@@ -59,60 +59,58 @@ class ServerConfig {
...
     static {
 
-        idleIntervalMillis = Long.getLong("sun.net.httpserver.idleInterval", DEFAULT_IDLE_INTERVAL_IN_SECS) * 1000;
-        if (idleIntervalMillis <= 0) {
-            idleIntervalMillis = DEFAULT_IDLE_INTERVAL_IN_SECS * 1000;
-        }
+        long providedIdleIntervalSecs = Long.getLong("sun.net.httpserver.idleInterval", DEFAULT_IDLE_INTERVAL_IN_SECS);
+        idleIntervalMillis = Math.multiplyExact(
+                providedIdleIntervalSecs > 0 ? providedIdleIntervalSecs : DEFAULT_IDLE_INTERVAL_IN_SECS,
+                1000);
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java:80]: That's a change of behaviour - maybe we should keep the original behaviour that was not unreasonable: if the result of the computation overflows, use the implementation default value.

Comment (vy) [reply src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java]: Fixed in 518f1eebe2ec0eb2a11f41853c301a61c16e1325.

File: src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java (hunk 1 above)
Comment (vy) [reply src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java]: Done.

**Pull request closed without merging**